language_service/src/completion.rs

```diff
@@ -765,105 +765,89 @@ fn callable_decl_to_completion_item(
     package_alias_from_manifest: &Option<Arc<str>>,
     callable_namespace: &qsc::hir::Idents,
     imports: &[ImportItem],
-    insert_open_at: Option<Range>,
+    insert_import_at: Option<Range>,
     indent: &String,
-) -> Option<(CompletionItem, SortPriority)> {
+) -> (CompletionItem, SortPriority) {
     let name = callable_decl.name.name.as_ref();
     // details used when rendering the completion item
     let detail = Some(display.hir_callable_decl(callable_decl).to_string());
     // Everything that starts with a __ goes last in the list
     let sort_group = u32::from(name.starts_with("__"));
 
-    // buffer to be filled up with an edit that would need to
-    // be applied if this completion item is selected
-    let mut additional_edits = vec![];
-
-    match &current_namespace_name {
-        // if there is no package alias for this callable (i.e. it
-        // is not the user package or stdlib) and
-        // if the current namespace is the same as the callable
-        // namespace
-        Some(curr_ns)
-            if package_alias_from_manifest.is_none()
-                && *curr_ns == Into::<Vec<_>>::into(callable_namespace) =>
-        {
-            None
-        }
-        _ => {
-            // calculate the qualification that goes before the import
-            // item
-            // if an exact import already exists, or if that namespace
-            // is glob imported, then there is no qualification
-
-            // If there is no matching import or glob import, then the
-            // qualification is the full namespace name
-
-            // an exact import is an import that matches the namespace
-            // and item name exactly
-            let namespace_as_strs = Into::<Vec<_>>::into(callable_namespace);
-            let preexisting_exact_import = imports.iter().any(|import_item| {
-                let import_item_namespace = &import_item.path[..import_item.path.len() - 1];
-                let import_item_name = import_item.path.last().map(|x| &**x);
-                *import_item_namespace == namespace_as_strs[..] && import_item_name == Some(name)
-            });
+    // calculate the qualification that goes before the import
+    // item
+    // if an exact import already exists, or if that namespace
+    // is glob imported, then there is no qualification
+
+    // If there is no matching import or glob import, then the
+    // qualification is the full namespace name
+
+    // an exact import is an import that matches the namespace
```

Comment:
nitpick: why the spaces between these comments? It seems like the could be all one paragraph with some punctuation between then. Not critical to functionality, of course.

Reply:
I was just using newlines to separate my sentences. I've updated it to use the more canonical periods-based sentence structure that English provides 😄